src/scale_limits_analyzer.rs

```diff
@@ -0,0 +1,157 @@
+use bluejay_core::definition::SchemaDefinition as CoreSchemaDefinition;
+use bluejay_core::AsIter;
+use bluejay_core::Directive;
+use bluejay_core::Value as CoreValue;
+use bluejay_core::{definition::prelude::*, ValueReference};
+use bluejay_parser::ast::{
+    definition::FieldDefinition,
+    definition::{DefaultContext, SchemaDefinition},
+    executable::ExecutableDocument,
+};
+use serde_json::Value;
+use std::collections::HashMap;
+
+pub type ScaleLimitsAnalyzer<'a> = bluejay_validator::executable::operation::Orchestrator<
+    'a,
+    ExecutableDocument<'a>,
+    SchemaDefinition<'a>,
+    serde_json::Map<String, serde_json::Value>,
+    ScaleLimits<'a>,
+>;
+
+pub struct ScaleLimits<'a> {
+    value_stack: Vec<Vec<&'a Value>>,
+    path_stack: Vec<&'a str>,
+    rates: HashMap<Vec<&'a str>, f64>,
+}
+
+impl<'a>
+    bluejay_validator::executable::operation::Visitor<
+        'a,
+        ExecutableDocument<'a>,
+        SchemaDefinition<'a>,
+        serde_json::Map<String, serde_json::Value>,
+    > for ScaleLimits<'a>
+{
+    type ExtraInfo = &'a Value;
+
+    fn new(
+        _operation_definition: &'a <ExecutableDocument as bluejay_core::executable::ExecutableDocument>::OperationDefinition,
+        _schema_definition: &'a SchemaDefinition<'a>,
+        _variable_values: &'a serde_json::Map<String, serde_json::Value>,
+        _cache: &'a bluejay_validator::executable::Cache<'a, ExecutableDocument, SchemaDefinition>,
+        extra_info: &'a Value,
+    ) -> Self {
+        Self {
+            value_stack: vec![vec![extra_info]],
+            path_stack: Vec::new(),
+            rates: Default::default(),
+        }
+    }
+
+    fn visit_field(
+        &mut self,
+        field: &'a <ExecutableDocument<'a> as bluejay_core::executable::ExecutableDocument>::Field,
+        field_definition: &'_ <SchemaDefinition as CoreSchemaDefinition>::FieldDefinition,
+        _scoped_type: bluejay_core::definition::TypeDefinitionReference<
+            '_,
+            <SchemaDefinition<'_> as CoreSchemaDefinition>::TypeDefinition,
+        >,
+        _included: bool,
+    ) {
+        self.path_stack.push(field.response_key());
+        let rate = Self::rate_for_field_definition(field_definition);
+        let values = self.value_stack.last().unwrap();
+        let mut nested_values = Vec::new();
+        values.iter().for_each(|value| {
+            let value_for_field = match value {
+                Value::Object(object) => object.get(field.response_key()),
+                Value::Null => None,
+                _ => panic!("invalid value"),
```

Comment:
We shouldn't panic here. We could either silently ignore this by doing:
```
_ => None,
```
or record an error and propagate it to the user